lib/tdf3/src/tdf.ts

```diff
@@ -960,8 +1018,8 @@ export async function decryptStreamFrom(
   const encryptedSegmentSizeDefault = defaultSegmentSize || DEFAULT_SEGMENT_SIZE;
 
   // check if the TDF is a legacy TDF
-  const specVersion = manifest.schemaVersion || manifest.tdf_spec_version;
-  const isLegacyTDF = !specVersion || specVersion.startsWith('3.');
+  const specVersion = manifest.schemaVersion || manifest.tdf_spec_version || '4.2.2';
```

Comment:
the `|| '4.2.2` looks out of place, it will be possible this is undefined. Make sure isTargetSpecLegacy can handle undefined spec

Reply:
Resolved, `4.2.2` was changed to `tdfSpecVersion` constant defined in the SDK